### PR TITLE
Add support for products/extensions in WORKAROUND_MODULES

### PR DESCRIPTION
**Pull request by alvarocarvajald**
This adds support to `WORKAROUND_MODULES` so it can include products (such as sles, sled, sles_sap) or extensions (such as ha) in the variable with the purpose of using it when skipping SCC
registration or testing in staging areas.

Also included in the PR is the inclusion of `sapapp` in the map of modules' short names to full names defined in lib/registration.pm.

- Related ticket: N/A
- Needles: N/A
- Verification run: http://mango.suse.de/tests/231 & http://mango.suse.de/tests/232

### CODE REVIEW

File: products/sle/main.pm [hunk 1]
```diff
@@ -278,7 +278,9 @@ if (sle_version_at_least('15') && !check_var('SCC_REGISTER', 'installation')) {
         my $addonurl;
 
         for my $short_name (@modules) {
-            my $full_name = $registration::SLE15_MODULES{$short_name};
+            # Map the module's full name from its short name.
+            # If it's not defined in $registration::SLE15_MODULES, then assume it's a Product/Extension and treat accordingly
+            my $full_name = $registration::SLE15_MODULES{$short_name} ? $registration::SLE15_MODULES{$short_name} : uc $short_name;
```
Comment (rwx788) [inline products/sle/main.pm]: What about creating hash for SLE15_PRODUCTS and put products there, and then we know if the key is from that hash it's a product and another repo name should be used. Then we can have some flag like is_product or is_module. I'm just afraid that this code will explode =)

Comment (alvarocarvajald) [reply products/sle/main.pm]: Thought about it, but currently all products are already using short names but in all caps (Product-SLES, Product-SLED, Product-SLES_SAP, Product-HPC and Product_WE) so decided against it. I agree it could be useful in the long run, but for now the map would be like `{ 'sles' => 'SLES', 'ha' => 'HA' }`, etc., so it seemed redundant to me.

Comment (rwx788) [reply products/sle/main.pm]: Good point. But what's about having flag? We check if it's in hash, if not it means it's product. Then we can use it in L#296 instead of another string comparison. WDYT?

Comment (alvarocarvajald) [reply products/sle/main.pm]: @rwx788 regarding is_product/is_module, you're thinking about having the methods in lib/registration.pm along with SLE15_MODULES and the tentative SLE15_PRODUCTS?

Comment (rwx788) [reply products/sle/main.pm]: I guess as of now it's fine to have it here in the method. If you think it can be useful somewhere else, we could indeed move it to registration module, that's up to you.

Comment (alvarocarvajald) [reply products/sle/main.pm]: @rwx788 I went with an `is_module` method in lib/registration.pm next to `%SLE15_MODULES` and `%SLE15_DEFAULT_MODULES` (makes sense to have all related module stuff in the same place, right?), and changes in L#283 and L#296 in main.pm to use the method. Currently doing some test runs, once they complete will push.

**Comment by alvarocarvajald** [comment]
Pushed changes. New test runs in: http://mango.suse.de/tests/235 & http://mango.suse.de/tests/236

File: products/sle/main.pm [hunk 2]
```diff
@@ -288,6 +290,10 @@ if (sle_version_at_least('15') && !check_var('SCC_REGISTER', 'installation')) {
                 $prefix .= "-Staging:" . get_var("STAGING");
             }
             my $module_repo_name = get_var("REPO_SLE${version}_MODULE_${repo_name}", "$prefix-Module-$full_name-POOL-$arch-Build$build-Media1");
+            # REPO_SLE* settings and repo names are different for products and modules
+            # Assign the proper repo name if current $short_name is not a module.
+            $module_repo_name = get_var("REPO_SLE_${repo_name}${version}_POOL", "$prefix-Product-$full_name-POOL-$arch-Build$build-Media1")
```
Comment (asmorodskyi) [inline products/sle/main.pm]: hard to follow your logic in such form . suggesting the following 

```
my $repo_variable_name= is_module($short_name) ? "REPO_SLE_${repo_name}${version}_POOL" : "REPO_SLE${version}_MODULE_${repo_name}";
my $module_repo_name = get_var($repo_variable_name, "$prefix-Module-$full_name-POOL-$arch-Build$build-Media1") ;
```

Comment (asmorodskyi) [reply products/sle/main.pm]: just a side note : we really overcomplicate things here . we have a variable which should define a repo name and we deciding about name of variable which should decide about name of a repo :\ . 
Not directly related to PR just thinking loud

Comment (rwx788) [reply products/sle/main.pm]: That's why we have this is_module method now, which makes it readable. I would rewrite these lines not to define variable twice and not with one liners as string is too long. But other than that looks good. 
As a side note, this thing was not supposed to stay too long, that's why it has WORKAROUND in the name of variable. So we may refactor it next time we introduce some changes.

Comment (asmorodskyi) [reply products/sle/main.pm]: REPO_SLE${version}_MODULE_${repo_name} and REPO_SLE_${repo_name}${version}_POOL  
 doesn't have WORKAROUND in their names , so will logic selecting variable which select something will remain after you will remove workaround ? I have impression that yes . My point is that instead selecting which variable will select something we can just simply select something :D

Comment (rwx788) [reply products/sle/main.pm]: Line #272. Part which you are referring to is parsing that variable and uses repo variables not to have hardcoded paths. You are talking about long term plan. You can create a ticket for refactoring. I don't think it's needed for adding product via ftp.

Comment (alvarocarvajald) [reply products/sle/main.pm]: @asmorodskyi working on suggestion. will push after test runs complete.

**Comment by alvarocarvajald** [comment]
New test runs: http://mango.suse.de/tests/239 & http://mango.suse.de/tests/240

**Comment by asmorodskyi** [comment]
LGTM
